pkg/crds/crds.go

```diff
@@ -18,10 +18,174 @@ limitations under the License.
+	if *(p.labelPathConfig.GroupVersionKind) == *config.GroupVersionKind {
+		p.labelPathConfig.Path = append(p.labelPathConfig.Path, config.Path...)
+	} else {
+		p.labelPathConfig = config
+	}
+}
+
+func (p *pathConfigs) AddAnnotationPathConfig(config transformers.PathConfig) {
+	if *(p.annotationPathConfig.GroupVersionKind) == *config.GroupVersionKind {
+		p.annotationPathConfig.Path = append(p.labelPathConfig.Path, config.Path...)
+	} else {
+		p.annotationPathConfig = config
+	}
+}
+
+func (p *pathConfigs) AddNamereferencePathConfig(config transformers.ReferencePathConfig) {
+	p.namereferencePathConfigs = transformers.MergeNameReferencePathConfigs(p.namereferencePathConfigs, config)
+}
+
+func (p *pathConfigs) AddPrefixPathConfig(config transformers.PathConfig) {
+	if *(p.prefixPathConfig.GroupVersionKind) == *config.GroupVersionKind {
+		p.prefixPathConfig.Path = append(p.prefixPathConfig.Path, config.Path...)
+	} else {
+		p.prefixPathConfig = config
+	}
+}
+
 // RegisterCRDs parse CRD schemas from paths and update various pathConfigs
 func RegisterCRDs(loader loader.Loader, paths []string) error {
+	pathConfigs := []pathConfigs{}
+	for _, path := range paths {
+		pathConfig, err := registerCRD(loader, path)
+		if err != nil {
+			return err
+		}
+		pathConfigs = append(pathConfigs, pathConfig...)
+	}
+	addPathConfigs(pathConfigs)
 	return nil
 }
+
+// register CRD from one path
+func registerCRD(loader loader.Loader, path string) ([]pathConfigs, error) {
+	result := []pathConfigs{}
+
+	content, err := loader.Load(path)
+	if err != nil {
+		return result, err
+	}
+
+	var types map[string]common.OpenAPIDefinition
+	if strings.HasSuffix(path, "json") {
+		err = json.Unmarshal(content, &types)
+	} else {
+		err = yaml.Unmarshal(content, &types)
+	}
+
+	crds := getCRDs(types)
+	for _, crd := range crds {
+		crdPathConfigs := pathConfigs{}
+		err = getCRDPathConfig(types, crd, crd, []string{}, &crdPathConfigs)
+		if err != nil {
+			return result, err
+		}
+		result = append(result, crdPathConfigs)
+	}
+
+	return result, nil
+}
+
+// getCRDs get all CRD types
+func getCRDs(types map[string]common.OpenAPIDefinition) []string {
+	crds := []string{}
+
+	for typename, t := range types {
+		properties := t.Schema.SchemaProps.Properties
+		if _, ok := properties["kind"]; ok {
+			if _, ok := properties["apiVersion"]; ok {
+				if _, ok := properties["metadata"]; ok {
+					crds = append(crds, typename)
+				}
+			}
+		}
+	}
+	return crds
+}
+
+func getCRDPathConfig(types map[string]common.OpenAPIDefinition, atype string, crd string, path []string, configs *pathConfigs) error {
+	if _, ok := types[crd]; !ok {
+		return nil
+	}
+
+	for propname, property := range types[atype].Schema.SchemaProps.Properties {
+		_, annotate := property.Extensions.GetString(Annotation)
+		if annotate {
+			configs.AddAnnotationPathConfig(
+				transformers.PathConfig{
+					CreateIfNotPresent: false,
+					GroupVersionKind:   &schema.GroupVersionKind{Kind: crd},
```

Comment:
IMO this doesn't construct a normal GVK.
We can build GVK from `apiVersion` and `kind`. You can use `TypeMeta{}.GroupVersionKind()`.
`apiVersion` and `kind` info can be find in your `getCRDs` method.